supabase/functions/clippy-search/index.ts

```diff
@@ -0,0 +1,194 @@
+import { serve } from 'https://deno.land/std@0.170.0/http/server.ts'
+import 'https://deno.land/x/xhr@0.2.1/mod.ts'
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.5.0'
+import GPT3Tokenizer from 'https://esm.sh/gpt3-tokenizer@1.1.5'
+import { Configuration, OpenAIApi } from 'https://esm.sh/openai@3.1.0'
+
+const openAiKey = Deno.env.get('OPENAI_KEY')
+const supabaseUrl = Deno.env.get('META_SUPABASE_URL')
+const supabaseServiceKey = Deno.env.get('META_SUPABASE_SERVICE_KEY')
+
+export const corsHeaders = {
+  'Access-Control-Allow-Origin': '*',
+  'Access-Control-Allow-Headers': 'authorization, x-client-info, apikey, content-type',
+}
+
+serve(async (req) => {
+  // Handle CORS
+  if (req.method === 'OPTIONS') {
+    return new Response('ok', { headers: corsHeaders })
+  }
+
+  if (!openAiKey) {
+    return new Response(
+      JSON.stringify({
+        error: 'Missing environment variable OPENAI_KEY',
+      }),
+      {
+        status: 400,
+        headers: { ...corsHeaders, 'Content-Type': 'application/json' },
+      }
+    )
+  }
```

Comment:
Ideally these error responses can be abstracted to a single place (eg. `try`-`catch`) in the future rather than lots of duplication.